### PR TITLE
Do not pass the option "device" in rsyslog.conf by default when syslog server's source address is configured

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it

An in-band syslog server will not receive any syslog if it is configured without a VRF specified, which is because `eth0` is always specified as the `device` of a syslog server and the syslog packets will be sent to `eth0` regardless of its destination IP address.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Pass the option "device" in rsyslog.conf only if when syslog server's source address is configured with a non-default VRF

#### How to verify it

Manually test:
1. Configuring a syslog server without VRF specified or with `default` as the VRF: no `device` passed in `rsyslog.conf`
2. Configuring a syslog server with non-default VRF: the configured VRF passed as `device` in `rsyslog.conf`

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by StormLiangMS** [comment]
hi @qiluo-msft could you have expert to take a look at this PR? And there is an internal conflict.

**Comment by qiluo-msft** [comment]
Is it covered by sonic-mgmt testcases?

**Comment by stephenxs** [comment]
> Is it covered by sonic-mgmt testcases?

I do not think so. this was found by our internal QA test.

**Comment by stephenxs** [comment]
> > > Is it covered by sonic-mgmt testcases?
> > 
> > 
> > I do not think so. this was found by our internal QA test.
> 
> Could you add a test for the behavior "an in-band syslog server will not receive any syslog if it is configured without a VRF specified"?

I'm not sure whether there is infra for such test cases. So, I need to check the details and get back to you
Is it possible to merge it first because it's a blocker to our 202305 release and then we will add the test case later?
Thanks

File: files/image_config/rsyslog/rsyslog.conf.j2 [hunk 1]
```diff
@@ -107,19 +107,24 @@ $RepeatedMsgReduction on
 {% set regex = conf.get('filter_regex') -%}
 
 {% set fmodifier = '!' if filter == 'exclude' else '' %}
-{% set device = 'eth0' if vrf == 'default' else vrf -%}
+{% set device = vrf if vrf != '' and vrf != 'default' -%}
```
Comment (qiluo-msft) [inline files/image_config/rsyslog/rsyslog.conf.j2:110]: >if [](http://example.com/codeflow?start=20&length=2)

There is no `else` after this `if`. Will it break?

**Comment by qiluo-msft** [comment]
@prsunny Could you help review?

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 1 above)
Comment (stephenxs) [reply files/image_config/rsyslog/rsyslog.conf.j2:110]: Hi
No.
If the condition `vrf != '' and vrf != 'default'` isn't satisfied, `device` is empty, which is expected.
In line [121](https://github.com/sonic-net/sonic-buildimage/pull/17616/files#diff-b5c3b035d28139ed77348578faf3d18d55884f61a98bb218fca31c993725a98eR121), the `device` is empty and option will be empty as well.
We do not pass `device` without `vrf` specified as a non-default one, which is the motivation of this PR.
This case is covered by the 1st manual test item `Configuring a syslog server without VRF specified or with default as the VRF: no device passed in rsyslog.conf`

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
ADO 27353460
@StormLiangMS Please help backport.

**Comment by StormLiangMS** [comment]
hi @stephenxs  @wen587  pls have test with 202305 on brcm/mellanox/cisco/marvell platform, would like to avoid regression.

**Comment by stephenxs** [comment]
> hi @stephenxs @wen587 pls have test with 202305 on brcm/mellanox/cisco/marvell platform, would like to avoid regression.

@StormLiangMS this is platform independent feature. It was tested on 202305 on mellanox platform.

**Comment by wen587** [comment]
Hi @StormLiangMS , I tested the rsyslog.conf generation in 202305 PROD. I tried with one MGMTdevice and one non-MGMTdevice patching the PR.
Both devices show the same difference as below: 
 
`action(type="omfwd" Target="some_ip" Port="514" Protocol="udp" Device="eth0")`
Changed to 
`action(type="omfwd" Target="some_ip" Port="514" Protocol="udp" Template="SONiCFileFormat")`
Since SONiCFileFormat is the default template. So it just eliminates the Device variable. There should be no regression.

**Pull request merged**